connector_prestashop/unit/importer.py

```diff
@@ -492,7 +493,12 @@ def _after_import(self, binding):
 def import_batch(session, model_name, backend_id, filters=None, **kwargs):
     """ Prepare a batch import of records from PrestaShop """
     backend = session.env['prestashop.backend'].browse(backend_id)
-    env = backend.get_environment(model_name, session=session)
+    connector_env_options = kwargs.get('connector_env_options', False)
+    if connector_env_options:
+        env = backend.get_environment(
+            model_name, session=session, **connector_env_options)
+    else:
+        env = backend.get_environment(model_name, session=session)
```

Comment:
For one question, I am trying to synchronize a shop with two stores, so one store has an url 'domain.com/shop1/api' and the other one has another url 'domain.com/shop2/api'. When a product import job is queued by the second url and has dependencies or after import do another models, the jobs queued by this dendencies stores the kwards 'shop_url', by this way when this job start pass the kwargs and in my custom environment update the api url to access.
If i not propagate kwards when the main importers creates new jobs, this starts through backend.location url and i want connect by the url store in job.

For second, Do you mean that i modify class ConnectorEnvironment in the connector module?